torch/jit/annotations.py

```diff
@@ -119,6 +120,11 @@ def get_type_line(source):
     type_lines = list(filter(lambda line: type_comment in line[1], lines))
 
     if len(type_lines) == 0:
+        type_pattern = re.compile('#[\t ]*type[\t ]*:')
```

Comment:
I don't know if we necessarily want to support prefixes that MyPy considers invalid. It might be better to just nudge a user in the right direction, so they fix their prefix to work with mypy.

Reply:
This regex is a little confusing, won't it also match `# type:`?

Reply:
it does, but had there been `# type:` we wouldn't have even run this regex due to this check : `if len(type_lines) == 0:`

Reply:
it's valid to have multiple type lines, so it's not enough to check that if there is one valid match. Can we instead filter all lines that include `type`, check for the exact match, and if there isn't an exact match check for the regex?

Reply:
In this particular case, we would still get `Number of type annotations (1) did not match the number of function parameters (2):`. We could tweak that error message to advise our user to check their type annotation prefixes.

Reply:
What about
```
def foo(
   a, # type: int
   b #     type: bad type annotation
)
# type: (...) -> int
```

Reply:
As long as there's **at least** one valid type annotation prefix, we should get an error. 
I just double checked that this exact example also gives the same error.

```python
@torch.jit.script
def add_two(a,# type: int
            b #     type: int
           ):
    # type: (...) -> int
    return a+b
```